src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -20952,32 +20952,1366 @@ public class C
+    {
+        field1.ToString();
+        field2.ToString(); // 1
+        local();
+
+        void local()
+        {
+            init();
+
+            [MemberNotNull(nameof(field1), nameof(field2))]
+            bool init() => throw null!;
+        }
+    }
+}
+""", MemberNotNullAttributeDefinition]);
+
+            c.VerifyDiagnostics(
+                // (10,9): warning CS8602: Dereference of a possibly null reference.
+                //         field2.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "field2").WithLocation(10, 9));
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/56256")]
+        public void MemberNotNull_StaticLocalFunction_WithinInstanceMethod()
+        {
+            var c = CreateNullableCompilation(["""
+using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    public string field1 = "";
+    public string? field2;
+
+    public void M()
+    {
+        init();
+        field1.ToString();
+        field2.ToString(); // 1
+
+        [MemberNotNull(nameof(field1), nameof(field2))]
+        static bool init() => throw null!;
+    }
+}
+""", MemberNotNullAttributeDefinition]);
+
+            c.VerifyDiagnostics(
+                // (11,9): warning CS8602: Dereference of a possibly null reference.
+                //         field2.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "field2").WithLocation(11, 9));
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/56256")]
+        public void MemberNotNull_StaticLocalFunction_WithinStaticMethod()
+        {
+            var c = CreateNullableCompilation(["""
+using System.Diagnostics.CodeAnalysis;
+public static class C
+{
+    public static string field1 = "";
+    public static string? field2;
+    public static string field3 = "";
+    public static string? field4;
+
+    public static void M()
+    {
+        init();
+        field1.ToString();
+        field2.ToString();
+        field3.ToString();
+        field4.ToString(); // 1
+
+        [MemberNotNull(nameof(field1), nameof(field2))]
+        static bool init() => throw null!;
+    }
+}
+""", MemberNotNullAttributeDefinition]);
+
+            c.VerifyDiagnostics(
+                // (15,9): warning CS8602: Dereference of a possibly null reference.
+                //         field4.ToString(); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "field4").WithLocation(15, 9));
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/56256")]
+        public void MemberNotNull_StaticLocalFunction_WithinInstanceMethod_StaticFields()
+        {
+            var c = CreateNullableCompilation(["""
+using System.Diagnostics.CodeAnalysis;
+public class C
+{
+    public static string field1 = "";
+    public static string? field2;
+
+    public void M()
+    {
+        init();
+        field1.ToString();
+        field2.ToString(); // 1
```

Comment:
>field2.ToString(); // 1 [](http://example.com/codeflow?start=8&length=23)

Is this warning expected? (It looks like we don't report a warning for the corresponding static method case.)